packages/SwingSet/src/controller/controller.js

```diff
@@ -303,22 +311,14 @@ export async function makeSwingsetController(
    * @returns { Promise<BundleID> }
    */
   async function validateAndInstallBundle(bundle, allegedBundleID) {
-    // TODO: validation: unpack, parse sources, check hashes
-
-    // during the transition to endo's new format, preemptively ignore the
-    // hash it provides
-    bundle = harden({
-      moduleFormat: bundle.moduleFormat,
-      endoZipBase64: bundle.endoZipBase64,
-    });
-
-    // this only computes the hash of the compartment map, it does not check
-    // that the rest of the bundle matches
-    const bundleID = await computeBundleID(bundle);
-    if (allegedBundleID) {
+    await checkBundle(bundle, computeSha512);
```

Comment:
This is meant to be the main defensive entry point, so we must assume `bundle` can be anything. We don't need to `harden` the object first, because `checkBundle` asserts `frozen`. But.. what if `bundle` has a getter for `endoZipBase64Sha512` that returns the bundle byte's hash the first time, then the caller's expected hash the second time? Could that trick it?

(this would be a `check-bundle` issue that I failed to think of when reviewing that other package, not a bug in this callsite, but we shouldn't switch to `checkBundle` until we know it's fully defensive)

Reply:
https://github.com/endojs/endo/issues/1106